model-optimizer/extensions/back/LayoutChangeForEinsum.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from extensions.ops.einsum import Einsum
+from mo.back.replacement import BackReplacementPattern
+from mo.graph.graph import Graph
+
+
+class LayoutChangeForEinsum(BackReplacementPattern):
+    """
+    The transformation adjusts Einsum equation to NCHW layout.
+    Subscripts for tensor of rank greater than three must be adjusted
+    to NCHW layout, meaning a label for the last dimension is moved
+    to the second position in the subscript.
+    There is an exception when the last label in the subscript is ellipsis
+    and covers multiple dimensions. In this case subscript is not changed and
+    Transpose to get original NHWC layout back is inserted.
+    The transformation is only applicable to TensorFlow case.
+    """
+    enabled = True
+    force_shape_inference = True
+    graph_condition = [lambda graph: graph.graph['fw'] == 'tf' and graph.graph['layout'] != 'NCHW']
```

Comment:
Actually my original comment was wrong. The "graph.graph['layout']" is always equal to the "NCHW" (it is set to this value during the layout change between middle and back phases). So this condition is always false.

Reply:
Fixed.